### PR TITLE
Chassis Class Refactor

### LINKED ISSUE #28: [BUG] Logger System Unused
#### Observed Behavior
The debugger system is unused. 

#### Expected Behavior
Making mistakes such as trying to move the chassis prior to calibrating should be logged appropriately.

#### How to Reproduce
1. Configure chassis
2. Move chassis without calibrating it

#### Project Info
Pros Kernel: 3.8.0
LemLib version: 0.4.5
Additional Libraries: None

#### Errors / Screenshots / Code
None

#### Additional Notes
Some examples of when the logger should be used:

1. Turning to face a point very close to the robot [Warning]
2. V5 device command failed to run successfully [Error]
3. Chassis sensor calibration failed [Error]
4. Odom configuration error (e.g 2 tracking wheels in same position) [Error]
5. Failed to read SD card [Error]
6. Improperly formatted file [Error]
7. Same motors on left and right side drivetrain [Error]

and many more situations that should be addressed in a PR

### PR DESCRIPTION
**Pull request by liam-teale**
#### Summary
Refactored the chassis class to use the logger, check ports, and overall cleanliness

#### Motivation
This is in preparation for builder patterns. Having existing code as clean as possible will make this large change easier

#### Test Plan (set logger mode to debug)
- [ ] Check error messages
- [ ] Check that port checks log errors
- [ ] Check that unplugging motors or sensors does not break odom

#### Additional Notes
Functionally: Added checks, small updates to logger
Internal API: slight changes
User-facing API: moved util functions into the `util` namespace

### CODE REVIEW

**Comment by BattleCh1cken** [comment]
I've run the following tests:

- [x] Test that correct values are being output
- [x] Try calibrating the chassis twice and see if it breaks the odom task
- [x] Try motion algs

Motion algs and output work correctly, but calibrating twice causes all output (x,y and theta) to be nan.

Edit: I have not tested any of this with three wheel odom, only with an imu and two tracking wheels.

**Comment by BattleCh1cken** [comment]
I've run the following tests:
- [x] calibrate the chassis twice without preserving position
- [x] calibrate the chassis twice while preserving the position
In both cases the odom task continued to work normally, and in the second case it kept the prior state, as expected. 

Edit:
These tests were also conducted:
- [x] calibrated chassis twice, and then used the moveTo method
- [x] calibrated the chassis twice while preserving position, and use the moveTo method

The moveTo method behaved as expected.

**Comment by liam-teale** [comment]
Excellent, thank you for your effort

File: include/lemlib/chassis/chassis.hpp [hunk 1]
```diff
@@ -101,8 +102,9 @@ class Chassis {
         /**
          * @brief Calibrate the chassis sensors
          *
+         * @param preservePose true if the pose should be preserved, false if not. False by default
```
Comment (Exortions) [inline include/lemlib/chassis/chassis.hpp:105]: A little questionable style-wise if 0 should be OK, but not a huge deal.

**Review by Exortions — APPROVED** [review]
Looks good to me. Although I didn't do a thorough review, I just briefly skimmed over all changes. If no more bugs are being reported, I'd say it is suitable for merging.

**Comment by liam-teale** [comment]
the data abort exception was _not_ fixed apparently

**Comment by liam-teale** [comment]
Closing. Many aspects of this PR have changed and it will be easier to start over

**Pull request closed without merging**